pilot/pkg/model/context.go

```diff
@@ -884,14 +890,39 @@ func (node *Proxy) SetWorkloadLabels(env *Environment) {
 
 // DiscoverIPMode discovers the IP Versions supported by Proxy based on its IP addresses.
 func (node *Proxy) DiscoverIPMode() {
+	var wildCardIPs, localHostIPs []string
 	if networkutil.AllIPv4(node.IPAddresses) {
 		node.ipMode = IPv4
+		wildCardIPs = append(wildCardIPs, constants.WildcardAddress)
+		localHostIPs = append(localHostIPs, constants.LocalhostAddress)
 	} else if networkutil.AllIPv6(node.IPAddresses) {
 		node.ipMode = IPv6
+		wildCardIPs = append(wildCardIPs, constants.WildcardIPv6Address)
+		localHostIPs = append(localHostIPs, constants.LocalhostIPv6Address)
 	} else {
 		node.ipMode = Dual
+		wildCardIPs = append(wildCardIPs, []string{constants.WildcardAddress, constants.WildcardIPv6Address}...)
+		localHostIPs = append(localHostIPs, []string{constants.LocalhostAddress, constants.LocalhostIPv6Address}...)
 	}
 	node.GlobalUnicastIP = networkutil.GlobalUnicastIP(node.IPAddresses)
+	node.wildCardIPs = wildCardIPs
+	node.localHostIPs = localHostIPs
```

Comment:
Hi @hzxuzhonghu, in fact they are not caching same,  the content for `wildCardIPs`  and `localHostIPs` depends on the service deployment, it can be, for example to `localHostIPs`, ["127.0.0.1"], ["::1"] or ["127.0.0.1", "::1"].

BTW, 
1. if not store them in proxy, then Istio must calculate them,
2. if not calculate them, then must store them
One of them should be choose.

Reply:
use 2 map to maintain `wildCardIPs` and `localHostIPs`